Source/WebCore/css/StyleRule.h

```diff
@@ -406,6 +409,8 @@ class StyleRuleScope final : public StyleRuleGroup {
     // Author written selector lists
     CSSSelectorList m_originalScopeStart;
     CSSSelectorList m_originalScopeEnd;
+    // Pointer to the owner StyleSheetContent to find the implicit scope (when there is no <scope-start>)
+    const StyleSheetContents* m_styleSheetOwner { nullptr };
```

Comment:
I don't see you clearing this anywhere, you can probably leave a dangling pointer behind. This should really be a smart pointer (WeakPtr probably, StyleSheetContents should be CanMakeWeakPtr).

Reply:
Fixed ✅